src/providers/dime/common/crypto.c

```diff
@@ -620,6 +620,7 @@ _generate_ed25519_keypair(void)
         PUSH_ERROR_OPENSSL();
         _secure_wipe(result, sizeof(ED25519_KEY));
         free(result);
+        result = NULL;
```

Comment:
On line 615, if the expression `if (RAND_bytes_d(result->private_key,sizeof(result->private_key)) != 1)` is TRUE and control flow enters that basic block, then I believe what you're saying is that RET_ERROR_PTR returns some value (presumably NULL) and control flow exits the function. The point here is that if control flow does indeed go this way, then lines 627 (`ed25519_publickey...`) through 629 (`return result;`) are never executed. Moreover, the function cannot return a dangling pointer this way because line 629 (`return result;`) is never executed. If all this is true, then yes, I believe [cppcheck](https://github.com/danmar/cppcheck) is wrong. Nice catch!